api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java

```diff
@@ -742,20 +741,60 @@ public CohortDefinition getPatientsWhoInitiatedPreTarvDuringCurrentMonthAndScree
     cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
     cd.addParameter(new Parameter("endDate", "End Date", Date.class));
     cd.addParameter(new Parameter("location", "Location", Location.class));
-    CohortDefinition tb = getPatientScreenedForTb();
 
-    String mappings = "onOrAfter=${startDate},locationList=${location}";
+    String mappings = "startDate=${startDate},endDate=${endDate},location=${location}";
     cd.addSearch(
-        "Pop",
+        "A2",
         map(
-            getPatientsWhoInitiatedPreTarvAtAfacilityDuringCurrentMonthC1(),
-            "startDate=${startDate},endDate=${endDate},location=${location}"));
-    cd.addSearch("TB", map(tb, mappings));
+            getPatientsWhoInitiatedPreTarvAtAfacilityDuringCurrentMonthA2(),
+            mappings));
+    cd.addSearch("TB", map(tbScreening(), mappings));
+
+    cd.setCompositionString("A2 AND TB");
+    //cd.setCompositionString("TB");
 
-    cd.setCompositionString("Pop AND TB");
 
     return cd;
   }
+  
+  private CohortDefinition tbScreening() {
```

Comment:
@cfaife I suggest to change the name of this function to something more clear to what it is doing...
```suggestion
  private CohortDefinition getNumberOfPatientTbScreenedInFirstEncounter()() {
```

Reply:
@cfaife And please move to tb queries or generic

Reply:
done